### DISCUSSION TITLE
Add support for border radius customization on rings

### DISCUSSION THREAD

**Discussion by hanna-del-rae** [tailwindlabs/tailwindcss#11143] [Ideas]
Our team came across a scenario where a the designs required a pill shaped button to maintain a square(ish)  focus ring.  Since the rings maintain the shape of the buttons by default, we could not use the built-in Tailwind ring for handing our focus states. Would it be possible to add support to give developers control over the border radius of the rings, rather than have to manually add and style a :before pseudo-element in such instances?

**Comment by lewis374** [comment]
Pleased to see your issue as I think I have the same problem, I am also confused and in need of light on this same issue. Need help.
[Fragomen Connect Client Portal](https://www.fragomenconnect.com/)

**Comment by andymccray** [comment]
As a workaround you could wrap the button contents in an additional container and style them to look like a pill: https://play.tailwindcss.com/N7rOyuKpTC

**Reply by tobistudio** [reply]
Really useful data. Thanks

**Comment by harrisono132123** [comment]
For me using 'rounded-none' before whatever border-radius variable you want (i.e. rounded-l-md), solved this for me.
